### DISCUSSION TITLE
Datapane is now unmaintained

### DISCUSSION THREAD

**Discussion by mands** [datapane/datapane#397] [Announcements]
Hi all,

We've made the difficult decision to stop driving this project and therefore we will no longer actively respond to issues or pull requests. If you would like to take over maintaining this project independently, please let us know so we can add a link to your forked project here.

The final release, `0.17.0`, has been optimized to focus on free local report saving, with unused features and analytics removed, and libraries updated to allow usage for as long as possible.

Thank You.

**Comment by Paul-Computomics** [comment]
That's unfortunate. We decided to use Datapane for internal reporting just a few days ago. Is there any chance that this project will be take over by somebody?

**Comment by Benrdn** [comment]
That's a bummer to hear. I bet a lot of places could use Datapane.

Quick question: Since we still rely on Datapane's CDN to open reports, do you guys plan on keeping it running? I'm guessing it costs money, so I'm kinda worried it might shut down someday.

**Reply by jocelyn-LC** [reply]
@Benrdn FYI the cdn was turned off. They were able to set up a dedirect to my Github repo. Let me know if you need specific versions to be supported.

**Comment by antdking** [comment]
Hi @Benrdn, we've no plans to turn the CDN off.

You can also specify a different location to source statics from by specifying the environment variable `DATAPANE_CDN_BASE=https://example.org/dp-statics/` when saving your reports.

**Reply by Benrdn** [reply]
Great to hear.
I think I didn't understand the second part, do you mean doing that so the source files will be pulled from a different location? 
If yes - is there a simple way to download all the files needed?

**Reply by antdking** [reply]
Correct;
With `DATAPANE_CDN_BASE=https://example.org/dp-statics`, if the statics are served from that domain + path, the report will fetch from them.


You can either build web-components yourself (`cd web-components; npm install; npm build`), or download the files listed below:
```
https://datapane-cdn.com/v0.17.0/assets/iframeResizer.contentWindow.map
https://datapane-cdn.com/v0.17.0/assets/iframeResizer.contentWindow.min.js
https://datapane-cdn.com/v0.17.0/base/index.es.js
https://datapane-cdn.com/v0.17.0/base/style.css
https://datapane-cdn.com/v0.17.0/report/Bokeh.68e4db27.es.js
https://datapane-cdn.com/v0.17.0/report/DataTable.a4a10952.es.js
https://datapane-cdn.com/v0.17.0/report/Formula.7d92138d.es.js
https://datapane-cdn.com/v0.17.0/report/Plotly.5452aeeb.es.js
https://datapane-cdn.com/v0.17.0/report/Vega.cf4ab013.es.js
https://datapane-cdn.com/v0.17.0/report/arrow-utils.24f569fd.es.js
https://datapane-cdn.com/v0.17.0/report/index.33215d7f.es.js
https://datapane-cdn.com/v0.17.0/report/index.css
https://datapane-cdn.com/v0.17.0/report/index.effbeeae.es.js
https://datapane-cdn.com/v0.17.0/report/index.es.js
https://datapane-cdn.com/v0.17.0/report/resize-observer.946ecf99.es.js
https://datapane-cdn.com/v0.17.0/report/tailwind.461ebf99.es.js
https://datapane-cdn.com/v0.17.0/report/tailwind.css
https://datapane-cdn.com/v0.17.0/template-components/Modal.ce.es.js
https://datapane-cdn.com/v0.17.0/template-components/SearchQuery.ce.es.js
https://datapane-cdn.com/v0.17.0/template-components/index.es.js
https://datapane-cdn.com/v0.17.0/template-components/index.es2.js
https://datapane-cdn.com/v0.17.0/vue.esm-browser.js
https://datapane-cdn.com/v0.17.0/vue.esm-browser.prod.js
```

For context, the default value for `DATAPANE_CDN_BASE` is `https://datapane-cdn.com/v0.17.0`.

**Reply by joannakraw** [reply]
Hi @antdking! 

I was wondering if the DataPane CDN was turned off? As for today, all of my previous DataPane reports stopped working and I cannot access this page https://datapane-cdn.com/v0.17.0/report/index.es.js.

**Reply by jocelyn-LC** [reply]
I rebuilt the v0.17.0 source files and uploaded them on github/jsdelivr.
You can replace the following in the report HTML source files: 
https://cdn.jsdelivr.net/gh/jocelyn-LC/dpane/v0.17.0/report/index.css
https://cdn.jsdelivr.net/gh/jocelyn-LC/dpane/v0.17.0/report/tailwind.css
https://cdn.jsdelivr.net/gh/jocelyn-LC/dpane/v0.17.0/report/index.es.js

I'm assuming that the instructions above would work to produce the files directly with this mirror CDN by setting the following environment variable `DATAPANE_CDN_BASE=https://cdn.jsdelivr.net/gh/jocelyn-LC/dpane/v0.17.0`

**Reply by ninoseki** [reply]
(I don't like spammy PR and so I don't do something like this again but) I forked Datapane and make it work with Pandas v2, Polars, etc. 
https://github.com/ninoseki/arakawa 
You may want to try it.

**Reply by joannakraw** [reply]
Thank you @ninoseki, I've tried your repository and it's amazing - it totally saved all of my previous reports and work!

**Reply by RWMostert** [reply]
Thanks @ninoseki!
We'll be switching our reports to Arakawa.  This is a fantastic tool, and we can't find a good alternative.

**Reply by mands** [reply]
Speaking on behalf of the old Datapane organization,

- Yes the CDN was turned off the other day as part of shutting down the remaining cloud resources for the company, we kept it going as long as we could but had to shut it down
- Unfortunately I didn't realise that the CDN was hosted by our cloud provider, and so previous reports would have stopped working

Going forwards,
- I have set up a redirect that means that all older reports accessing the old CDN at https://datapane-cdn.com redirect to the newer CDN at https://cdn.jsdelivr.net/gh/jocelyn-LC/dpane (thanks @jocelyn-LC for uploading!)
- I do recommend anyone wanting to build new reports to use the fork by @ninoseki at https://github.com/ninoseki/arakawa - it includes lots of new features and pandas v2 support!

Thanks for everyone who has used to product, I hope it was helpful!

**Reply by felixschlesinger** [reply]
Thanks @mands and @jocelyn-LC 

I see the re-direct and the resources are available at that new URL. However with the re-direct from the old (datapane-cdn.com) URL I am getting a security error now:
![image](https://github.com/user-attachments/assets/1ccd5128-4ab0-4347-88ff-426eb8082594)
 (this is for local HTML file, but similar error for one hosted on a remote webserver).

I think this needs a setting update on the new hosting location?

**Reply by mands** [reply]
Updated - should work now - thanks!

**Reply by MickeyGT** [reply]
Thank you very much for your work on this. I can't explain in words how much I'd appreciate it if you could also host the CDN for previous versions as well as we have reports on versions 15 and above that no longer work. 

> I rebuilt the v0.17.0 source files and uploaded them on github/jsdelivr. You can replace the following in the report HTML source files: https://cdn.jsdelivr.net/gh/jocelyn-LC/dpane/v0.17.0/report/index.css https://cdn.jsdelivr.net/gh/jocelyn-LC/dpane/v0.17.0/report/tailwind.css https://cdn.jsdelivr.net/gh/jocelyn-LC/dpane/v0.17.0/report/index.es.js
> 
> I'm assuming that the instructions above would work to produce the files directly with this mirror CDN by setting the following environment variable `DATAPANE_CDN_BASE=https://cdn.jsdelivr.net/gh/jocelyn-LC/dpane/v0.17.0`

**Reply by ninoseki** [reply]
I think the easiest way is using standalone mode which produces a report without relying on CDN assets.

```py
import datapane as dp

dp.Report(dp.Text("Hello, world!")).save(path="report.html", open=True, standalone=True)
```

This trick works with v0.15.x. If you use v0.16.x, downgrading/upgrading the version can be a workaround.

**Reply by MickeyGT** [reply]
Thank you for the recommendation. On the generation side, we've moved over to arakawa to generate offline reports with the standalone option. The problem is with existing reports on my colleagues' computers. They can no longer visualize the existing reports until the CDN for the older versions is also up and running.

**Reply by ninoseki** [reply]
Sorry I misunderstand the context.
Yes, existing reports need CDN redirection to be rendered and I can do nothing for that. 

@mands Just a friendly confirmation, do you have a plan to re-host CDN assets (`<0.17`) or redirect requests to somewhere?

**Reply by jocelyn-LC** [reply]
I'd be happy to regenerate and push to my repo assets for other versions, I'm hosting v0.17.0 and v0.15.7 right now which should make the reports work out of the box. Please reply with precise version(s) needed. @MickeyGT

**Reply by MickeyGT** [reply]
@jocelyn-LC  Hello and thank you again for your work on this. Versions 15.5 and 16.7 are the ones we used previously. If I can help in any way with building/uploading those versions or others, let me know. If we can get all versions from 15 and up available on your CDN, I think we can cover support for almost all existing datapane reports(globally, not just within my organization)

**Reply by jocelyn-LC** [reply]
@MickeyGT, it should work now. The created files for v0.15.5 are quite large. Hopefully everything still works properly.

**Reply by MickeyGT** [reply]
Can confirm reports created on v0.15.5 now work fine again. 🥳🎉

**Reply by ValentinLRT** [reply]
@jocelyn-LC Hello, could you do it for the v0.15.3 ? All our reports are in this version and stopped working as the others.

**Reply by jocelyn-LC** [reply]
@ValentinLRT Just uploaded. You can try and let me know if there's any issue :)

**Reply by ValentinLRT** [reply]
@jocelyn-LC They work fine again. Thank you !

**Comment by Buroni** [comment]
I'd be happy to help with future maintenance of the open source library. @mands I have a fork at https://github.com/Buroni/datapane if you'd like to add a link. I've enjoyed developing local reports over the last few years and would love to see the feature continue to be used!

**Reply by mands** [reply]
Hey @Buroni - sounds good, feel free to push any changes back here and we can coordinate a release as part of a more community-led project. I think there was some useful work to aid local reports that were in a branch or two somewhere, e.g. encryption, that are worth porting across if you still have them.

**Comment by mscolnick** [comment]
I want to express my admiration and respect for the Datapane team for their contributions. This library has made a significant impact in low-code Python application/dashboard community.

We are building something similar https://github.com/marimo-team/marimo that may be a viable alternative for those interested. We are in the early days, but moving fast. We don't integrate directly with Jupyter and instead provide our own kernel, which gives us the ability to support some of our defining features.

**Reply by mands** [reply]
Thanks for the kind words @mscolnick and good luck with the Marimo project, looks great!

**Comment by isaacn-lab** [comment]
Hello,
This is very sad, we use `datapane` in multiple projects, and it's served us well so far. So thank you very much for the effort!
Our plan is to continue using it as long as there are no severe compatibility issues. 

At the moment, I am a bit concerned with `pandas = ">=1.4.0,<2.0.0"` constraint. I believe many projects would require the latest pandas soon (if not already). Is this constraint hard to remove?

Thank you!
-Nikita

**Reply by jmrichardson** [reply]
It appears that pandas 2.2.0 has an issue.  I think its fonts related or something with how its presenting.  Dont know how to fix it but got it working again with pandas==2.1.4.  Unfortunately, lots of other repos are already requiring > 2.2.0

Can anyone point me into how to fix this.  Will work on a patch if possible

**Comment by christianmatei** [comment]
I will be happy to work on this, from a pure hobby perspective. My link https://github.com/christianmatei/datapane Thanks.

**Comment by ninoseki** [comment]
Hello Datapane team,
(Cc @mands) 

First of all, I'd like to say big thanks to you for creating a great library, datapane.

I believe it's very useful and it's worth keep maintaining.  Thus I’m thinking about creating a fork of the project and the following is the WIP version of my fork.

https://github.com/ninoseki/arakawa/tree/wip

Notes:
- Forked with renaming.
- I don’t intend to get any profits from the fork.
- All the trademarks will be removed. (WIP)
- Cloud based features (upload, analysts, etc.) will be removed. (WIP)
- CDN will be replaced with jsDelivr. (WIP)

I think, according to the license (Apache License 2.0), the forking is okay but I want to make sure whether you have a comment or objection about this fork.
I will stop working on this fork if you have any concerns or negative impressions. So please let me know if you have any.

---

Note: I also sent the same email to info@datapane.com but there is no response. So let me try to use this discussion thread.

**Reply by mands** [reply]
Hi @ninoseki - thanks for the message and kind words! Yep, no problem, please go ahead and fork away, happy to see the project and codebase live on!

**Reply by ninoseki** [reply]
Thanks!
I published the first version yesterday and it can work along with newer libraries such as Pandas v2, NumPy v2, etc.
https://github.com/ninoseki/arakawa

**Reply by mands** [reply]
Amazing - great to hear!
